types/steam-tradeoffer-manager/lib/classes/TradeOffer.d.ts

```diff
@@ -2,6 +2,31 @@ import TradeOfferManager = require("../../index");
 import SteamID = require("steamid");
 import CEconItem = require("steamcommunity/classes/CEconItem");
 
+interface BaseItem {
+    appid: number;
+    contextid: number;
+    amount?: number | undefined;
+}
+
+interface ItemWithId extends BaseItem {
+    id: string;
+    assetid?: string | number | undefined;
+}
+
+interface ItemWithAssetId extends BaseItem {
+    id?: string | undefined;
+    assetid: string | number;
+}
+
```

Comment:
I did it like this because you can either define the id or the assetid, but you have to define atleast one of them. So this is more accurate than making the id and assetid both optional. See this check https://github.com/DoctorMcKay/node-steam-tradeoffer-manager/blob/3652b13b14988d9f954f38d744f2046307a682b6/lib/classes/TradeOffer.js#L254

Reply:
Sorry, that's not what I meant. I meant that assetId and id should have the same types because they are the same. So if assetId is `string | number | undefined`, then id should be as well.

Reply:
Ah I get it. I will change the assetid and id type to string | number. This is the same as what the steamcommunity package says it can be. I guess I could even reuse their types for the item type.